tests/acl/test_acl.py

```diff
@@ -3,796 +3,768 @@
+    pytest.mark.topology("any")
 ]
 
 BASE_DIR = os.path.dirname(os.path.realpath(__file__))
-DUT_TMP_DIR = os.path.join('tmp', os.path.basename(BASE_DIR))
-FILES_DIR = os.path.join(BASE_DIR, 'files')
-TEMPLATE_DIR = os.path.join(BASE_DIR, 'templates')
+DUT_TMP_DIR = "acl_test_dir"  # Keep it under home dir so it persists through reboot
+FILES_DIR = os.path.join(BASE_DIR, "files")
+TEMPLATE_DIR = os.path.join(BASE_DIR, "templates")
+
+ACL_TABLE_TEMPLATE = "acltb_table.j2"
+ACL_RULES_FULL_TEMPLATE = "acltb_test_rules.j2"
+ACL_RULES_PART_TEMPLATES = tuple("acltb_test_rules_part_{}.j2".format(i) for i in xrange(1, 3))
+ACL_REMOVE_RULES_FILE = "acl_rules_del.json"
+
+DEFAULT_SRC_IP = "20.0.0.1"
+
+DOWNSTREAM_DST_IP = "192.168.0.2"
+DOWNSTREAM_IP_TO_ALLOW = "192.168.0.4"
+DOWNSTREAM_IP_TO_BLOCK = "192.168.0.8"
 
-ACL_TABLE_TEMPLATE = 'acltb_table.j2'
-ACL_RULES_FULL_TEMPLATE = 'acltb_test_rules.j2'
-ACL_RULES_PART_TEMPLATES = tuple('acltb_test_rules_part_{}.j2'.format(i) for i in xrange(1, 3))
-ACL_REMOVE_RULES_FILE = 'acl_rules_del.json'
+UPSTREAM_DST_IP = "192.168.128.1"
+UPSTREAM_IP_TO_ALLOW = "192.168.136.1"
+UPSTREAM_IP_TO_BLOCK = "192.168.144.1"
 
-DST_IP_TOR = '192.168.0.1'
-DST_IP_TOR_FORWARDED = '192.168.8.1'
-DST_IP_TOR_BLOCKED = '192.168.16.1'
-DST_IP_SPINE = '192.168.128.1'
-DST_IP_SPINE_FORWARDED = '192.168.136.1'
-DST_IP_SPINE_BLOCKED = '192.168.144.1'
+VLAN_BASE_MAC_PATTERN = "72060001{:04}"
 
-LOG_EXPECT_ACL_TABLE_CREATE_RE = '.*Created ACL table.*'
-LOG_EXPECT_ACL_TABLE_REMOVE_RE = '.*Successfully deleted ACL table.*'
-LOG_EXPECT_ACL_RULE_CREATE_RE = '.*Successfully created ACL rule.*'
-LOG_EXPECT_ACL_RULE_REMOVE_RE = '.*Successfully deleted ACL rule.*'
+LOG_EXPECT_ACL_TABLE_CREATE_RE = ".*Created ACL table.*"
+LOG_EXPECT_ACL_TABLE_REMOVE_RE = ".*Successfully deleted ACL table.*"
+LOG_EXPECT_ACL_RULE_CREATE_RE = ".*Successfully created ACL rule.*"
+LOG_EXPECT_ACL_RULE_REMOVE_RE = ".*Successfully deleted ACL rule.*"
 
 
-@pytest.fixture(scope='module')
+@pytest.fixture(scope="module")
 def setup(duthost, tbinfo, ptfadapter):
+    """Gather all required test information from DUT and tbinfo.
+
+    Args:
+        duthost: A fixture to interact with the DUT.
+        tbinfo: A fixture to gather information about the testbed.
+
+    Yields:
+        A Dictionary with required test information.
+
     """
-    setup fixture gathers all test required information from DUT facts and tbinfo
-    :param duthost: DUT host object
-    :param tbinfo: fixture provides information about testbed
-    :return: dictionary with all test required information
-    """
+    topo = tbinfo["topo"]["type"]
 
-    tor_ports = []
-    spine_ports = []
-    tor_ports_ids = []
-    spine_ports_ids = []
-    port_channels = []
-    acl_table_ports = []
+    if topo not in ("t0", "t1"):
+        pytest.skip("ACL test not supported on topology: \"{}\"".format(tbinfo["topo"]["name"]))
```

Comment:
couple of minor things:
1. you could use pytest_require() here.
2. is this change dualtor ready? dualtor's topology type is 't0'

Reply:
Good catch! Added.